primitives/src/circuit/merkle_tree.rs

```diff
@@ -10,26 +10,32 @@
 
 use crate::{
     circuit::rescue::RescueGadget,
-    merkle_tree::{AccMemberWitness, MerklePath, MerkleTree, NodePos, NodeValue},
+    merkle_tree::{
+        examples::RescueMerkleTree, internal::MerkleNode, MerkleTreeScheme, ToTraversalPath,
+    },
     rescue::RescueParameter,
 };
 use ark_ec::TEModelParameters as Parameters;
 use ark_ff::PrimeField;
-use ark_std::{vec, vec::Vec};
+use ark_std::vec::Vec;
 use jf_relation::{errors::CircuitError, BoolVar, Circuit, PlonkCircuit, Variable};
 
-#[derive(Clone)]
-struct MerkleNodeBooleanEncoding<F: PrimeField> {
-    sibling1: NodeValue<F>,
-    sibling2: NodeValue<F>,
+type NodeVal<F> = <RescueMerkleTree<F> as MerkleTreeScheme>::NodeValue;
+type MerkleProof<F> = <RescueMerkleTree<F> as MerkleTreeScheme>::MembershipProof;
```

Comment:
Is it better to call it `MembershipProof<F>`? As we might support non-membership proofs in the future for sparse Merkle tree.

Reply:
Sure, good call.

Reply:
@chancharles92 Also for `MerkleProofVar` -> `MembershipProofVar` and the like, right?